src/utils/utf8.c

```diff
@@ -152,24 +188,54 @@ utf8_toupper(utf8_ctx_t *ctx, const char *s, char *buf, size_t buflen)
 	return len;
 }
 
+
+ssize_t
+utf8_subs_diacritics(utf8_ctx_t *ctx, strbuf_t *buf)
+{
+	UErrorCode ec = U_ZERO_ERROR;
+	uint16_t buf1[MAX_BUFFER_SIZE];
+	int32_t len = buf->bufsize;
```

Comment:
This is not correct:
- The length of the string is `buf->length` rather than `buf->bufsize` (this is the buffer size which may be arbitrarily larger).
- It would actually be cleaner to take the return value of the utf8_to_utf16() function instead, as it explicitly returns the number of units produced in this conversion.